reporters/GlobalSummaryReporter.js

```diff
@@ -0,0 +1,237 @@
+    this.priceFeed = priceFeed;
+
+    this.web3 = this.empEventClient.web3;
+
+    this.empContract = this.empEventClient.emp;
+
+    this.formatDecimalString = createFormatFunction(this.web3, 2, 4);
+
+    this.collateralSymbol = "DAI";
+    this.syntheticSymbol = "ETHBTC";
+  }
+
+  update = async () => {
+    await this.empClient.update();
+    await this.empEventClient.update();
+    await this.priceFeed.update();
+  };
+
+  generateSummaryStatsTable = async () => {
+    await this.update();
+
+    // 1. Sponsor stats table
+    console.group();
+    console.log(bold("Sponsor summary stats"));
+    console.log(italic("- Collateral deposited counts collateral transferred into contract from creates and deposits"));
+    console.log(
+      italic(
+        "- Collateral withdrawn counts collateral transferred out of contract from withdrawals, redemptions, expiry settlements, liquidation reward withdrawals, and fees paid"
+      )
+    );
+    console.log(italic("- Tokens minted counts synthetic tokens created"));
+    console.log(italic("- Tokens repaid counts synthetic tokens burned via redemptions and expiry settlements"));
+
+    await this._generateSponsorStats();
+    console.groupEnd();
+
+    // 2. Tokens stats table
+    // TODO
+
+    // 3. Liquidation stats table
+    // TODO
+
+    // 4. Dispute stats table
+    // TODO
+
+    // 5. DVM stats table
+    // TODO
+  };
+
+  _generateSponsorStats = async () => {
+    const { toBN, toWei } = this.web3.utils;
+
+    const newSponsorEvents = this.empEventClient.getAllNewSponsorEvents();
+    const depositEvents = this.empEventClient.getAllDepositEvents();
+    const createEvents = this.empEventClient.getAllCreateEvents();
+    const withdrawEvents = this.empEventClient.getAllWithdrawEvents();
+    const redeemEvents = this.empEventClient.getAllRedeemEvents();
+    const liquidationRewardEvents = this.empEventClient.getAllLiquidationWithdrawnEvents();
+    const expirySettlementEvents = this.empEventClient.getAllSettleExpiredPositionEvents();
+    const regularFeeEvents = this.empEventClient.getAllRegularFeeEvents();
+    const finalFeeEvents = this.empEventClient.getAllFinalFeeEvents();
+
+    const currentBlockNumber = Number(await this.web3.eth.getBlockNumber());
+    const blockNumberOneDayAgo = currentBlockNumber - (await this._getLookbackTimeInBlocks(24 * 60 * 60));
+
+    let allSponsorStatsTable = {};
+
+    // - Lifetime # of unique sponsors.
+    const uniqueSponsors = {};
+    const currentUniqueSponsors = this.empClient.getAllPositions();
+    for (let event of newSponsorEvents) {
+      uniqueSponsors[event.sponsor] = true;
+    }
+    allSponsorStatsTable["# of unique sponsors"] = {
+      cumulative: Object.keys(uniqueSponsors).length,
+      current: Object.keys(currentUniqueSponsors).length
+    };
+
+    // - Cumulative collateral deposited into contract: Deposits, Creates
+    let collateralDeposited = toBN("0");
+    let collateralDepositedDaily = toBN("0");
```

Comment:
nit: can we also change the language here to just refer to it as a period rather than "daily"?  For instance, we could change this variable to `collateralDepositedPeriod` or `collateralDepositedWithinPeriod`.

Reply:
done